src/model.py

```diff
@@ -2002,79 +1969,62 @@ def init_components(self, subtasks):
 
 
         #######################################
-        # Base components
+        # Task-specific CNN layers
         #######################################
         components = torch.nn.ModuleDict({
-            task: torch.nn.ModuleDict({
-                # final outputs for each task
-                # in features are concats of embeddings, shared layers, and task cnns
-                "linear": self.linear_block(in_features=cnn_dim*3, out_features=3),
-
-                # map subtask information back to shared hidden state size
-                "re_encode": self.linear_block(
-                    in_features=cnn_dim*2,  # after attention
-                    out_features=cnn_dim
-                )
-            })
+            task: torch.nn.ModuleDict({})
```

Comment:
Create empty ModuleDict, instead of `linear` and `re_encode`.
These were both refactored to where they are used 
(`linear` -> `linear_{stack}`; `re_encode` -> `stack-relation-first_task-re_encode`)